src/main/java/net/mcreator/generator/Generator.java

```diff
@@ -257,15 +258,37 @@ public List<GeneratorFile> generateElement(GeneratableElement element, boolean f
 		// generate all source files
 		List<GeneratorTemplate> generatorTemplateList = getModElementGeneratorTemplatesList(element.getModElement(),
 				performFSTasks, element);
+		List<GeneratorTemplatesList> templateLists = getModElementListTemplates(element.getModElement(), performFSTasks,
+				element);
 		if (generatorTemplateList != null) {
 			for (GeneratorTemplate generatorTemplate : generatorTemplateList) {
 				String templateFileName = (String) ((Map<?, ?>) generatorTemplate.getTemplateData()).get("template");
 
 				Map<String, Object> dataModel = generatorTemplate.getDataModel();
 				extractVariables(generatorTemplate, dataModel);
 
-				String code = getTemplateGeneratorFromName("templates").generateElementFromTemplate(element,
-						templateFileName, dataModel, element.getAdditionalTemplateData());
+				String code = "";
+				if (generatorTemplate.isListTemplate() && templateLists != null) {

```

Comment:
This code could see some comments so we can understand how this works, the same for changes in GeneratorTokens

Reply:
✔️ Fixed (I think)